c/parallel/include/cccl/c/segmented_sort.h

```diff
@@ -0,0 +1,95 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA Core Compute Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#ifndef CCCL_C_EXPERIMENTAL
+#  error "C exposure is experimental and subject to change. Define CCCL_C_EXPERIMENTAL to acknowledge this notice."
+#endif // !CCCL_C_EXPERIMENTAL
+
+#include <cuda.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <cccl/c/extern_c.h>
+#include <cccl/c/types.h>
+
+CCCL_C_EXTERN_C_BEGIN
+
+typedef struct cccl_device_segmented_sort_build_result_t
+{
+  int cc;
+  void* cubin;
+  size_t cubin_size;
+  CUlibrary library;
+  cccl_type_info key_type;
+  cccl_type_info offset_type;
+  cccl_op_t large_segments_selector_op;
+  cccl_op_t small_segments_selector_op;
+  CUkernel segmented_sort_fallback_kernel;
+  CUkernel segmented_sort_kernel_small;
+  CUkernel segmented_sort_kernel_large;
+  CUkernel three_way_partition_init_kernel;
+  CUkernel three_way_partition_kernel;
+  void* runtime_policy;
+  void* partition_runtime_policy;
+  cccl_sort_order_t order;
+} cccl_device_segmented_sort_build_result_t;
```

Comment:
This struct has some holes (caused by padding). 

```
Type: int(i) sizeof = 4, alignof = 4
Type: CUkernel(P9CUkern_st) sizeof = 8, alignof = 8
Type: CUlibrary(P8CUlib_st) sizeof = 8, alignof = 8
Type: cccl_type_info(14cccl_type_info) sizeof = 24, alignof = 8
Type: cccl_op_t(9cccl_op_t) sizeof = 64, alignof = 8
Type: cccl_sort_order_t(17cccl_sort_order_t) sizeof = 4, alignof = 4
Type: void *(Pv) sizeof = 8, alignof = 8
```

If `int cc` and `cccl_sort_order_t order` were adjacent to one another, we could same 8 bytes off the size of `cccl_device_segmented_sort_build_result_t` if that is a concern.

Reply:
Good catch. Can we leave this for another PR? I can create an issue to track this and other c.parallel build structs

Reply:
Tracked here https://github.com/NVIDIA/cccl/issues/6456